src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
>QPC is allowed to use 32 of the 64 bits for one second.

If QPC uses 32 bits for one second, and assuming 1 microsecond resolution, the values returned by QPC will not increment by 1. They will increment by 4292 (= 2^32 / 1e6). Then you don't need to represent those increments with precision 1.  The loss of precision from the floating point representation is negligible compared to the clock resolution, in this example.

But whatever absolute values you pick, the precision of the floating point representation is 52 bits. That means you can divide 142 years into microseconds. On shorter (and more practical) time periods you can have precisions much higher than the internal clock resolution.